lib/taurus/core/tango/tangoattribute.py

```diff
@@ -811,7 +811,11 @@ def setRange(self, *limits):
             limits = limits[0]
         low, high = limits
         low = Quantity(low)
+        if low.dimensionless:
+            low = Quantity(low, self._units)
```

Comment:
I saw the travis report, but it is very rare, I could not reproduce the failures locally.

```
 python  -m unittest taurus.core.tango.test.test_tangoattribute.AttributeTestCase
MainThread     INFO     2017-02-02 16:08:45,569 Starter: Starting server TangoSchemeTest/unittest...
MainThread     INFO     2017-02-02 16:08:46,578 Starter: Server TangoSchemeTest/unittest has been started
MainThread     WARNING  2017-02-02 16:08:46,713 TaurusRootLogger: /home/cfalcon/tools/taurus.git/lib/taurus/core/tango/test/test_tangoattribute.py:814: DeprecationWarning: _get_w_value is deprecated since 4.0. Use .wvalue instead
  got = getattr(read_value, k)

MainThread     WARNING  2017-02-02 16:08:46,715 TaurusRootLogger: /home/cfalcon/tools/taurus.git/lib/taurus/core/tango/test/test_tangoattribute.py:814: DeprecationWarning: _get_value is deprecated since 4.0. Use .rvalue instead
  got = getattr(read_value, k)

..ss...............................MainThread     WARNING  2017-02-02 16:08:48,398 TaurusRootLogger: /home/cfalcon/tools/taurus.git/lib/taurus/core/tango/test/test_tangoattribute.py:802: DeprecationWarning: getUnit is deprecated since 4.0. Use .rvalue.units instead
  got = getattr(a, k)

MainThread     WARNING  2017-02-02 16:08:48,399 TaurusRootLogger: /home/cfalcon/tools/taurus.git/lib/taurus/core/tango/test/test_tangoattribute.py:814: DeprecationWarning: TangoAttrValue.type is deprecated since 4.0. Use TangoAttribute.type instead
  got = getattr(read_value, k)

MainThread     WARNING  2017-02-02 16:08:48,399 TaurusRootLogger: /home/cfalcon/tools/taurus.git/lib/taurus/core/tango/test/test_tangoattribute.py:814: DeprecationWarning: TangoAttrValue.unit is deprecated since 4.0. Use TangoAttribute.unit instead
  got = getattr(read_value, k)

.....................MainThread     INFO     2017-02-02 16:08:49,281 Starter: Hard killing server TangoSchemeTest/unittest...
MainThread     INFO     2017-02-02 16:08:49,288 Starter: Server TangoSchemeTest/unittest has been stopped
MainThread     INFO     2017-02-02 16:08:49,296 Starter: Deleted device TangoSchemeTest/unittest/temp-1
MainThread     INFO     2017-02-02 16:08:49,296 Starter: Hard killing server TangoSchemeTest/unittest...
MainThread     INFO     2017-02-02 16:08:49,296 Starter: Server TangoSchemeTest/unittest has been stopped
MainThread     INFO     2017-02-02 16:08:49,315 Starter: Deleted Server TangoSchemeTest/unittest

----------------------------------------------------------------------
Ran 56 tests in 3.801s

OK (skipped=2)

```
About the implementation, I agree that maybe is better to use "low.magnitude", but with my Pint  version (0.8.dev0) it is valid. I will try with the docker.

```
from taurus.external.pint import Q_
q = Q_(1)
q2 = Q_(q, 'mm')
 print q2 #1 mm

```